aws/structure.go

```diff
@@ -1088,6 +1088,51 @@ func flattenESClusterConfig(c *elasticsearch.ElasticsearchClusterConfig) []map[s
 	return []map[string]interface{}{m}
 }
 
+func expandESCognitoOptions(m map[string]interface{}) *elasticsearch.CognitoOptions {
```

Comment:
It seems like its possible to simply some of the resource logic by accepting the `[]interface{}` from the schema here instead of the `map[string]interface{}`. Then we can handle the length/nil checks in one place.

e.g.

```go
if v, ok := d.GetOk("cognito_options"); ok {
  input.CognitoOptions = expandESCognitoOptions(v.([]interface{}))
}
```